File: backend/app/domain/entities/vaccination.py
from dataclasses import dataclass
from datetime import date, datetime
from uuid import UUID


@dataclass
class Vaccination:
    id: UUID
    baby_id: UUID
    vaccine_name: str
    dose_number: int
    scheduled_date: date
    administered_date: date | None
    hospital_name: str | None
    memo: str | None
    created_at: datetime

File: frontend/src/features/vaccination/types/vaccination.ts
export interface Vaccination {
  id: string;
  babyId: string;
  vaccineName: string;
  doseNumber: number;
  scheduledDate: string;
  administeredDate: string | null;
  hospitalName: string | null;
  memo: string | null;
  isOverdue: boolean;
  daysUntil: number | null;
  createdAt: string;
}

export interface MarkAdministeredRequest {
  administeredDate: string;
  hospitalName?: string;
}
